### PR TITLE
Removes the map border

### PR DESCRIPTION
**Pull request by TemporalOroboros**
## About The Pull Request
Masks away the occlusion and lighting planes outside of the map bounds by applying a new plane that acts as a mask for the map in a way similar to the emissive lighting system. This allows the parallax plane to be visible byond the bounds of the map and vastly smooths out z-level transitions.

There are a few issues at the moment, which is why this is a draft.
TODO:
- [ ] Figure out a better map boundary for lavaland and the ice moon. Space doesn't fit the aesthetic. Might just go back to the void if I can't figure something else out.
- [ ] Auto-Z-Shift ~~ghosts~~ anything that reaches the map boundary. Right now they just come to a dead stop and it takes a sec to realize that the boundary is there.

## Why It's Good For The Game
map_border_delta

## Changelog

:cl:
qol: Space no longer looks pitch black if you manage to get far enough away from the station.
/:cl:

### CODE REVIEW

**Comment by Colovorat** [comment]
>  Figure out a better map boundary for lavaland and the ice moon. Space doesn't fit the aesthetic. Might just go back to the void if I can't figure something else out.

texture of infinite rocks?

also there should be at least some indication for ghosts where is the borders i suppose

**Comment by LemonInTheDark** [comment]
This doesn't totally fix the issues say, spacedrifting binoculars have, but it at least helps a bit. Ups the presentation of flying around as a ghost by like 200% tho

>Auto-Z-Shift ghosts anything that reaches the map boundary. Right now they just come to a dead stop and it takes a sec to realize that the boundary is there.

It's to some degree important that ghosts can find z level borders, I've found it useful in debugging once or twice.
Maybe only autoshift ghosts without the debugger perm? idk

File: code/__DEFINES/layers.dm [hunk 1]
```diff
@@ -100,7 +100,10 @@
 
 #define BLACKNESS_PLANE 0 //To keep from conflicts with SEE_BLACKNESS internals
 
-
+/// The map used to mask the plane.
```
Comment (LemonInTheDark) [inline code/__DEFINES/layers.dm:103]: what's "the plane"?

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

**Comment by Colovorat** [comment]
you could draw something like a golden line for ghosts where the edge of the map should be

File: code/__DEFINES/layers.dm (hunk 1 above)
Comment (TemporalOroboros) [reply code/__DEFINES/layers.dm:103]: I think I crossed map and plane in the doc. I'll fix it.

**Comment by timothymtorres** [comment]
It only took someone twenty years to finally do this... you sir are a legend!

**Comment by Time-Green** [comment]
Truly epic. 

Out of curiosity, will you be able to see the edge of the lava planet when on a z-level bordering the station z-level, or will it show the generic background?

**Comment by Time-Green** [comment]
> > Truly epic.
> > Out of curiosity, will you be able to see the edge of the lava planet when on a z-level bordering the station z-level, or will it show the generic background?
> 
> Currently it shows the space parallax at the edge of lavaland. Hence why it's still in draft. I'm working on a system for z-dependent parallax.

That'd be the most epic thing ever, maybe a bit too cool to be honest

**Comment by D4C-420** [comment]
>Currently it shows the space parallax at the edge of lavaland. Hence why it's still in draft.
>I'm working on a system for z-dependent parallax.

Can I suggest lava for the boundary on lava land. Would make it look like your on an island in a sea of lava plus most miners will probably stay away from the wall out of habit keeping the illusion.  ( maybe make the far right side have like 2 tiles worth of lava from the edge to sell the effect but that's outside the scope of this PR)

**Comment by TemporalOroboros** [comment]
I've encountered an issue where the occlusion blackness is not interacting with plane masters at all. Seems like a BYOND issue, and so long as it exists this will need to stay in draft.

**Comment by optimumtact** [comment]
>Masks away the occlusion and lighting planes outside of the map bounds by applying a new plane that acts as a mask for the map in a way similar to the emissive lighting system. This allows the parallax plane to be visible byond the bounds of the map and vastly smooths out z-level transitions.

I do not understand how this works. please provide more detail

**Comment by optimumtact** [comment]
please exlpain

**Comment by TemporalOroboros** [comment]
Sorry, classes are picking up and I don't have much time to iron out the existing issues this has. I'll reopen this when I have more time to dedicate to it.

**Comment by LemonInTheDark** [comment]
😭

**Pull request closed without merging**